app/jobs/check_catalogues_job.rb

```diff
@@ -22,9 +16,9 @@ def perform
     current_catalogue = Hash.new(0)
     Catalogue.all.each{|c| current_catalogue[c.id] = c.catalogue_num}
 
-    #scrape new posts from new catalogue
+    #scrape new posts from new catalogues
     catalogue_to_add = add_new_catalogue(current_catalogue, new_catalogues)
-    ScrapeCataloguesJob.perform_later(catalogue_to_add)
+    ScrapeCataloguesJob.perform_later(catalogue_to_add) if catalogue_to_add.length > 0

```

Comment:
`x.any?` is a little more readable than `x.length > 0`
```ruby
ScrapeCataloguesJob.perform_later(catalogue_to_add) if catalogue_to_add.any?
```
or you may prefer:
```ruby
ScrapeCataloguesJob.perform_later(catalogue_to_add) unless catalogue_to_add.empty?
```
Choose whatever one you find easier to understand.